app-arch/bzip2/bzip2-1.0.6-r10.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit flag-o-matic multilib-minimal toolchain-funcs
+
+DESCRIPTION="A high-quality data compressor used extensively by Gentoo Linux"
+HOMEPAGE="http://www.bzip.org/"
+SUSE_PV="${PV}.0.2"
+SUSE_P="${PN}-${SUSE_PV}"
+SRC_URI="http://ftp.suse.com/pub/people/sbrabec/bzip2/tarballs/${SUSE_P}.tar.gz"
+S="${WORKDIR}/${SUSE_P}"
+
+LICENSE="BZIP2"
+SLOT="0/1" # subslot = SONAME
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd"
+IUSE="static static-libs"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.0.6-progress.patch
+	"${FILESDIR}"/${PN}-1.0.4-POSIX-shell.patch #193365
+	"${FILESDIR}"/${PN}-1.0.6-mingw.patch #393573
+	"${FILESDIR}"/${PN}-1.0.6-CVE-2016-3189.patch #620466
+)
+
+DOCS=( manual.pdf CHANGES NEWS README.autotools README README.XML.STUFF )
+HTML_DOCS=( manual.html )
+
+multilib_src_configure() {
+	# use static && append-ldflags -static # doesn't seem to work...
+	# fails at gen_usr_ldscript because adding -static to ldflags
+	# prevents generation of libbz2.so
+
+	ECONF_SOURCE="${S}" econf \
+		$(use_enable static-libs static)
+}
+
+multilib_src_install() {
+	emake DESTDIR="${D%/}" install
+	gen_usr_ldscript -a bz2
```

Comment:
In previous revision we called this only for *multilib_is_native_abi* (see https://github.com/gentoo/gentoo/blob/master/app-arch/bzip2/bzip2-1.0.6-r9.ebuild#L78). Is it really correct to call it for each ABI?

...and while writing: This isn't the reason why *static* failed, is it?

Reply:
Well, if I use the methodology above to append -static to the thing, it fails
here because there is no shared library made.

Reply:
```
 * ERROR: app-arch/bzip2-1.0.6-r10::gentoo failed (install phase):
 *   unable to read SONAME from libbz2.so
 * 
 * Call stack:
 *     ebuild.sh, line  124:  Called src_install
 *   environment, line 2789:  Called multilib-minimal_src_install
 *   environment, line 2004:  Called multilib_foreach_abi 'multilib-minimal_abi_src_install'
 *   environment, line 2198:  Called multibuild_foreach_variant '_multilib_multibuild_wrapper' 'multilib-minimal_abi_src_install'
 *   environment, line 1891:  Called _multibuild_run '_multilib_multibuild_wrapper' 'multilib-minimal_abi_src_install'
 *   environment, line 1889:  Called _multilib_multibuild_wrapper 'multilib-minimal_abi_src_install'
 *   environment, line  408:  Called multilib-minimal_abi_src_install
 *   environment, line 1994:  Called multilib_src_install
 *   environment, line 2422:  Called gen_usr_ldscript '-a' 'bz2'
 *   environment, line 1262:  Called die
 * The specific snippet of code:
 *                       [[ -z ${tlib} ]] && die "unable to read SONAME from ${lib}";
 * 
```
Using the commented out logic for `USE=static` results in this. The only `*.so` files
in image are `usr/lib{,64}/libbz2.so` which are `broken symbolic link to libbz2.so.1.0.6`.